packages/orm/src/EntityManager.ts

```diff
@@ -1905,6 +1900,67 @@ export class EntityManager<C = unknown, Entity extends EntityW = EntityW, TX ext
       );
     }
   }
+
+  fork(): EntityManager<C, Entity, TX> {
+    const oldEm = this;
+    const pendingEntities = oldEm.entities.filter((e) => getInstanceData(e).pendingOperation !== "none");
+    if (pendingEntities.length > 0) {
+      fail("Cannot fork an EntityManager with pending changes");
+    }
+    // copy the context so that it's distinct between the two ems and so we can update any references from the old em
+    // to the new one later
+    const ctx: C = { ...oldEm.ctx };
+    const newEm = new EntityManager<C, Entity, TX>(ctx, { em: oldEm });
+    // hydrate (ie, create) each entity from the oldEm in the newEm
+    oldEm.#entitiesByTag.forEach((entities, tag) => {
+      // tags are always defined by the base constructor, so we use that as hydrate expects the base class as its argument
+      const baseCstr = getConstructorFromTag(tag);
+      // We copy the raw row data for each entity to avoid any potential conflict. Then we ensure the correct concrete
+      // __class key is set for hydrate to use to actually instantiate the entity.
+      const rows = entities.map((e) => {
+        const { row, data } = getInstanceData(e);
+        return { ...(row ?? data), __class: e.constructor.name };
+      });
+      newEm.hydrate(baseCstr, rows);
+    });
+    // set the preload cache for each loaded relation from oldEm in the newEm so that get / isLoaded should just work
+    // on the new entities.  group by concrete class so that we only have to gather the relation definitions once per
+    // type.
+    groupBy(oldEm.entities, (e) => e.constructor as EntityConstructor<Entity>).forEach((entities, cstr) => {
+      const meta = getMetadata(cstr);
+      // use allFields so that we get fields from the base type as well
+      const relations = Object.entries(meta.allFields)
+        // these are all the kinds for concrete (ie, not custom) relations
+        .filter(([, field]) => ["o2m", "lo2m", "m2o", "m2m", "o2o", "poly"].includes(field.kind))
+        .map(([name]) => name);
+      entities.forEach((oldEntity) => {
+        // each entity could have different loaded status for each relation
+        const loadedRelations = relations.filter((field) => (oldEntity as any)[field].isLoaded);
+        const cache = new Map(
+          loadedRelations.map((field) => [
+            field,
+            // this should be safe because we are only using loaded relations and we made sure to copy every entity between ems
+            toArray((oldEntity as any)[field].get).map((e: Entity) => newEm.#entitiesById.get(e.idTagged)!),
+          ]),
+        );
+        newEm.#preloadedRelations.set(oldEntity.idTagged, cache);
+        const newEntity = newEm.#entitiesById.get(oldEntity.idTagged)!;
+        // force each relation to preload on the new entity
+        loadedRelations.forEach((relation) => (newEntity as any)[relation].preload());
```

Comment:
Fwiw I was thinking we could skip explicitly calling `.preload()` from within `.fork()` itself, and just wait until the caller does/does not call the `.get` / `.isLoaded` methods on the relations, which would then lazily/just-in-time call `.preload()`.

Does that work out, or do we need to do this?

Reply:
none of the references explicitly call `preload` themselves from any `get` / `isLoaded` path.  It's only called from data loaders which are obviously async.  If that situation changes then we can remove this, but in the meantime i didn't want this PR to sprawl.

Reply:
Ah okay, I really thought the `.isLoaded`s probed `isPreloaded` but I guess not; agreed that could be a follow up.